lib/src/component_declaration/annotations.dart

```diff
@@ -224,13 +250,16 @@ class AbstractComponent2 implements AbstractComponent { // ignore: deprecated_me
 /// Props are declared as fields, which act as stubs for generated getters/setters that proxy Map key-value pairs.
 ///
 ///     @PropsMixin()
-///     abstract class GraultPropsMixin {
+///     mixin GraultPropsMixin on UiProps {
 ///       Map get props;
 ///
 ///       Object garply;
 ///     }
 ///
 /// Classes using this annotation must include the abstract `props` getter.
+///
+/// __NOTE:__ This is only required for legacy boilerplate and can be omitted
+/// for new implementations. However, it can still be used for custom configurations.
 class PropsMixin implements TypedMap {
```

Comment:
From offline discussion: Let's deprecate `@PropsMixin()` since they can just use `@Props()` if they need to use something like the `keyNamespace` argument.

```suggestion
///
/// __Deprecated.__ Use the `@Props()` annotation instead if you need to make use of an annotation argument. 
/// Otherwise, this can be removed completely. Will be removed in the 4.0.0 release of over_react.
@Deprecated('Use the @Props() annotation if you need to make use of an annotation argument. Otherwise, this can be removed completely. Will be removed in the 4.0.0 release of over_react.')
class PropsMixin implements TypedMap {
```

Reply:
That multi line suggestion tho